packages/admin/cms-admin/src/generator/future/generateForm/generateFormField.ts

```diff
@@ -1,12 +1,55 @@
 import { IntrospectionEnumType, IntrospectionInputValue, IntrospectionNamedTypeRef, IntrospectionObjectType, IntrospectionQuery } from "graphql";
 
-import { FormConfig, FormFieldConfig, isFormFieldConfig } from "../generator";
+import { Adornment, FormConfig, FormFieldConfig, isFormFieldConfig } from "../generator";
 import { camelCaseToHumanReadable } from "../utils/camelCaseToHumanReadable";
 import { findQueryTypeOrThrow } from "../utils/findQueryType";
 import { Imports } from "../utils/generateImportsCode";
 import { isFieldOptional } from "../utils/isFieldOptional";
 import { findFieldByName, GenerateFieldsReturn } from "./generateFields";
 
+type AdornmentData = {
+    adornmentString: string;
+    adornmentImport?: { name: string; importPath: string };
+};
+
+const getAdornmentData = ({ adornmentData }: { adornmentData: Adornment }): AdornmentData => {
+    let adornmentString = "";
+    let adornmentImport = { name: "", importPath: "" };
+
+    if (typeof adornmentData === "string") {
+        return { adornmentString: adornmentData };
+    }
+
+    if (typeof adornmentData.icon === "string") {
+        adornmentString = `<${adornmentData.icon}Icon />`;
+        adornmentImport = {
+            name: `${adornmentData.icon} as ${adornmentData.icon}Icon`,
+            importPath: "@comet/admin-icons",
+        };
+    } else if (typeof adornmentData.icon === "object") {
+        if ("import" in adornmentData.icon) {
+            adornmentString = `<${adornmentData.icon.name} />`;
+            adornmentImport = {
+                name: `${adornmentData.icon.name} as ${adornmentData.icon.name}Icon`,
+                importPath: "@comet/admin-icons",
+            };
+        } else {
+            const { name, ...iconProps } = adornmentData.icon;
+            adornmentString = `<${name}Icon
+                ${Object.entries(iconProps)
+                    .map(([key, value]) => `${key}="${value}"`)
+                    .join("\n")}
+            />`;
+            adornmentImport = {
+                name: `${adornmentData.icon.name} as ${adornmentData.icon.name}Icon`,
+                importPath: "@comet/admin-icons",
+            };
+        }
+    }
```

Comment:
We'll probably have the same if/else statements for the icons in static select, right? Could we create a function to share the code?

Reply:
Yes, I will reuse this function for the selects. Also, I would like to update the selects in their own PR, they seem a bit more complicated to me. I don't want to block this PR with them.